srcts/src/main.ts

```diff
@@ -4155,8 +4155,28 @@ function main(): void {
     }
 
     if (dep.script && !restyle) {
-      const scripts = $.map(asArray(dep.script), function (scriptName) {
-        return $("<script>").attr("src", href + "/" + encodeURI(scriptName));
+      const scripts_attrs = asArray(dep.script);
+      const scripts = $.map(scripts_attrs, function (x) {
+        let script = document.createElement("script");
+
+        // htmlDependency()'s script arg may be a list of attributes
+        if (typeof x === "string") {
+          x = { src: x };
+        }
+        const attrs = Object.keys(x);
+
+        for (let i = 0; i < attrs.length; i++) {
+          const attr = attrs[i];
+          let val = x[attr];
+
+          if (attr === "src") {
+            val = href + "/" + encodeURI(val);
+          }
+          // If val isn't truthy (e.g., null), consider it a boolean attribute
+          val ? script.setAttribute(attr, val) : script.toggleAttribute(attr);
+        }
```

Comment:
Leverage jquery for the for-loop;
`toggleAttribute()` is not IE friendly.

```suggestion
        $.map(attrs, function(val, attr) {

          if (attr === "src") {
            val = href + "/" + encodeURI(val);
          }
          // If val isn't truthy (e.g., NA), only set the key
          script.setAttribute(attr, val ? val : "")
        })
```

Reply:
https://caniuse.com/?search=toggleAttribute